starters/next/pages/collections/[handle].js

```diff
@@ -0,0 +1,186 @@
+import { useState } from 'react';
+import { useRouter } from 'next/router';
+import { nacelleClient } from 'services';
+import ProductCard from 'components/ProductCard';
+import styles from 'styles/Collection.module.css';
+
+function Collection(props) {
+  const router = useRouter();
+  const { collection } = props;
+  const [products, setProducts] = useState(props.products);
+  const [canFetch, setCanFetch] = useState(props.canFetch);
+  const [isFetching, setIsFetching] = useState(false);
+
+  const activeProducts = canFetch
+    ? products?.slice(0, products.length - 1)
+    : products;
+
+  // Performs a GraphQL query to Nacelle to get paginated products
+  // from a collection, using the last `nacelleEntryId` as a cursor.
+  // (https://nacelle.com/docs/querying-data/storefront-sdk)
+  const handleFetch = async () => {
+    setIsFetching(true);
+    const after = products[products?.length - 1].nacelleEntryId;
+    const { productCollections } = await nacelleClient.query({
+      query: PRODUCTS_QUERY,
+      variables: { handle: router.query.handle, after }
+    });
+    const newProducts = productCollections[0]?.products;
+    if (newProducts) {
+      setCanFetch(newProducts.length === 12);
+      setProducts([...products, ...newProducts]);
+    }
+    setIsFetching(false);
+  };
```

Comment:
@NWRichmond I don't think numbered pagination is going to work here.  If you look at Nuxt starter it has a load more button (since we don't know full length of response). I wonder if we can adopt the same pattern here - so we only fetch a minimal amount of items to start

Reply:
I tried to keep a lot of the features similar to what we are doing with the Nuxt starter, so this pagination should be the same as what we are doing with Nuxt unless something has changed.

Reply:
@jasonredick you're totally right.  I misunderstood Nick's comment.

Reply:
I think we need to revisit this for all starters. Shifting complexity / fetching to the build step is something we should promote over client-side fetching, unless there are explicit design requirements re: infinite scroll.